### ISSUE TITLE
[FEAT] Upgrade SMS Classifier Model with LSTM Layer

### ISSUE THREAD

**Issue by PrajwalMadhyastha** [PrajwalMadhyastha/Finlight-Android#121]
### Description

Our current SMS classification model in `train_sms_classifier.py` uses a `GlobalAveragePooling1D` layer. While this is efficient, it treats the SMS as a "bag of words," losing all information about word order and sentence structure. This limits its ability to understand the context of complex financial messages where the sequence of words is crucial.

This task is to upgrade the model architecture to better understand sequential data, which should lead to higher accuracy and fewer misclassifications.

### Proposed Solution

The `GlobalAveragePooling1D` layer should be replaced with a Recurrent Neural Network (RNN) layer, specifically an **LSTM (Long Short-Term Memory)** or **GRU (Gated Recurrent Unit)** layer.

An LSTM processes the SMS word-by-word, in order, allowing it to learn the grammatical patterns and contextual relationships that are vital for accurately classifying financial texts.

**File to Modify:** `train_sms_classifier.py`

**Current Architecture:**
```python
model = tf.keras.Sequential([
    tf.keras.layers.Input(shape=(MAX_SEQUENCE_LENGTH,), dtype=tf.int64),
    tf.keras.layers.Embedding(VOCAB_SIZE, EMBEDDING_DIM, name="embedding"),
    tf.keras.layers.GlobalAveragePooling1D(), # <-- This will be replaced
    tf.keras.layers.Dense(16, activation='relu'),
    tf.keras.layers.Dropout(0.5),
    tf.keras.layers.Dense(1, activation='sigmoid')
])
```

**Proposed Architecture:**
```python
model = tf.keras.Sequential([
    tf.keras.layers.Input(shape=(MAX_SEQUENCE_LENGTH,), dtype=tf.int64),
    tf.keras.layers.Embedding(VOCAB_SIZE, EMBEDDING_DIM, name="embedding"),
    tf.keras.layers.LSTM(64), # <-- New LSTM layer
    tf.keras.layers.Dense(16, activation='relu'),
    tf.keras.layers.Dropout(0.5),
    tf.keras.layers.Dense(1, activation='sigmoid')
])
```

### Acceptance Criteria
- [ ] The train_sms_classifier.py script is updated to replace the GlobalAveragePooling1D layer with an LSTM or GRU layer.
- [ ] The model successfully trains with the new architecture.
- [ ] The sms_classifier.tflite model and vocab.txt files are re-exported.
- [ ] The on-device instrumentation tests (SmsClassifierTest.kt) pass with the new model, confirming it runs correctly and maintains or improves classification accuracy on known test cases.
